runners/direct-java/src/main/java/org/apache/beam/runners/direct/BoundedReadEvaluatorFactory.java

```diff
@@ -83,7 +83,8 @@ public void cleanup() {}
       if (sourceEvaluators.putIfAbsent(transform, evaluatorQueue) == null) {
         // If no queue existed in the evaluators, add an evaluator to initialize the evaluator
         // factory for this transform
-        BoundedSource<OutputT> source = transform.getTransform().getSource();
+        Bounded<OutputT> bound = (Bounded<OutputT>) transform.getTransform();
```

Comment:
That fixes the above problem but then creates a new problem in ParDoSingleEvaluatorFactory and ParDoMultiEvaluatorFactory. 

```
AppliedPTransform<PCollection<InputT>, PCollection<OutputT>, Bound<InputT, OutputT>>
          application,

Bound mismatch: The type ParDo.Bound<InputT,OutputT> is not a valid substitute for the bounded parameter <TransformT extends PTransform<InputT,OutputT>> of the type AppliedPTransform<InputT,OutputT,TransformT>	

ParDoSingleEvaluatorFactory.java	/beam-runners-direct-java/src/main/java/org/apache/beam/runners/direct	line 76	Java Problem
```

and similar in ParDoMultiEvaluatorFactory.  That said, the only call into that method does a cast from 'AppliedPTransform<?, ?, ?>' anyway so any compile time checking is relatively bogus so maybe flipping the code in ParDoSingleEvaluatorFactory might be OK.    I'll try and dig more into that tomorrow, but any ideas would be appreciated.